test/NinjaTools.FluentMockServer.Tests/Models/Equality/EqualityTestBase.cs

```diff
@@ -0,0 +1,150 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using Force.DeepCloner;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using NinjaTools.FluentMockServer.Tests.TestHelpers;
+using NinjaTools.FluentMockServer.Tests.TestHelpers.Data.Random;
+using NinjaTools.FluentMockServer.Tests.TestHelpers.Data.TheoryData;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NinjaTools.FluentMockServer.Tests.Models.Equality
+{
+    public abstract class EqualityTestBase<T> : XUnitTestBase<T> where T : class
+    {
+        // ReSharper disable once StaticMemberInGenericType
+        protected static List<object[]> TestDataOverride { get; set; }
+        protected EqualityTestBase(ITestOutputHelper outputHelper) : base(outputHelper)
+        { }
+        
+        [Theory]
+        [MemberData(nameof(GetEqualityOperatorTestData))]
+        public void Equals_Should_Compare_For_Equality_By_Properties([NotNull] T a, [NotNull] T b, bool areEqual)
+        {
+            // Assert
+            switch (areEqual)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)